Content.Server/Temperature/Systems/EntityHeaterSystem.cs

```diff
@@ -44,71 +28,32 @@ public override void Update(float deltaTime)
             }
         }
     }
 
-    private void OnExamined(EntityUid uid, EntityHeaterComponent comp, ExaminedEvent args)
-    {
-        if (!args.IsInDetailsRange)
-            return;
-
-        args.PushMarkup(Loc.GetString("entity-heater-examined", ("setting", comp.Setting)));
-    }
-
-    private void OnGetVerbs(EntityUid uid, EntityHeaterComponent comp, GetVerbsEvent<AlternativeVerb> args)
-    {
-        if (!args.CanAccess || !args.CanInteract)
-            return;
-
-        var setting = (int) comp.Setting;
-        setting++;
-        setting %= SettingCount;
-        var nextSetting = (EntityHeaterSetting) setting;
-
-        args.Verbs.Add(new AlternativeVerb()
-        {
-            Text = Loc.GetString("entity-heater-switch-setting", ("setting", nextSetting)),
-            Act = () =>
-            {
-                ChangeSetting(uid, nextSetting, comp);
-                _popup.PopupEntity(Loc.GetString("entity-heater-switched-setting", ("setting", nextSetting)), uid, args.User);
-            }
-        });
-    }
-
     private void OnPowerChanged(EntityUid uid, EntityHeaterComponent comp, ref PowerChangedEvent args)
     {
         // disable heating element glowing layer if theres no power
         // doesn't actually turn it off since that would be annoying
         var setting = args.Powered ? comp.Setting : EntityHeaterSetting.Off;
-        _appearance.SetData(uid, EntityHeaterVisuals.Setting, setting);
+        Appearance.SetData(uid, EntityHeaterVisuals.Setting, setting);
     }
 
-    private void ChangeSetting(EntityUid uid, EntityHeaterSetting setting, EntityHeaterComponent? comp = null, ApcPowerReceiverComponent? power = null)
+    public override void ChangeSetting(Entity<EntityHeaterComponent?> heater, EntityHeaterSetting setting)
     {
-        if (!Resolve(uid, ref comp, ref power))
-            return;
-
-        comp.Setting = setting;
-        power.Load = SettingPower(setting, comp.Power);
-        _appearance.SetData(uid, EntityHeaterVisuals.Setting, setting);
-        _audio.PlayPvs(comp.SettingSound, uid);
+        ChangeSetting((heater, heater.Comp, null), setting);
     }
 
-    private float SettingPower(EntityHeaterSetting setting, float max)
+    public void ChangeSetting(Entity<EntityHeaterComponent?, ApcPowerReceiverComponent?> heater,
+        EntityHeaterSetting setting)
     {
-        switch (setting)
-        {
-            case EntityHeaterSetting.Low:
-                return max / 3f;
-            case EntityHeaterSetting.Medium:
-                return max * 2f / 3f;
-            case EntityHeaterSetting.High:
-                return max;
-            default:
-                return 0f;
-        }
+        base.ChangeSetting(heater, setting);
+        if (!Resolve(heater,ref heater.Comp2))
+            return;
+        heater.Comp2.Load = SettingPower(setting, heater.Comp1!.Power);
+        Appearance.SetData(heater, EntityHeaterVisuals.Setting, setting);
```

Comment:
Sussy null suppression because this can cause nullrefs and isn't guaranteed to exist.